connectors/sources/s3.py

```diff
@@ -117,36 +137,44 @@ async def _get_content(self, doc, region, timestamp=None, doit=None):
             )
             return
         logger.debug(f"Downloading {filename}")
-        async with self.client(region_name=region) as s3:
-            try:
-                resp = await s3.get_object(Bucket=bucket, Key=filename)
-                await asyncio.sleep(0)
-                file_content, chunk = BytesIO(), True
-                while chunk:
-                    chunk = await resp["Body"].read(MAX_CHUNK_SIZE) or b""
-                    file_content.write(chunk)
-                file_content.seek(0)
-                data = file_content.read()
-                file_content.close()
-                logger.debug(f"Downloaded {filename} for {doc['size_in_bytes']} bytes ")
-                return {
-                    "_timestamp": timestamp,
-                    "_attachment": get_base64_value(content=data),
-                    "_id": doc["id"],
-                }
-            except (ClientError, ServerTimeoutError, AioReadTimeoutError) as exception:
-                if (
-                    getattr(exception, "response", {}).get("Error", {}).get("Code")
-                    == "InvalidObjectState"
-                ):
-                    logger.warning(
-                        f"{filename} of {bucket} is archived and inaccessible until restored. Error: {exception}"
-                    )
-                else:
-                    logger.error(
-                        f"Something went wrong while extracting data from {filename} of {bucket}. Error: {exception}"
-                    )
-                    raise
+        document = {
+            "_id": doc["id"],
+            "_timestamp": doc["_timestamp"],
+        }
+        source_file_name = ""
+        try:
+            async with NamedTemporaryFile(mode="wb", delete=False) as async_buffer:
+                await s3_client.download_fileobj(
+                    Bucket=bucket, Key=filename, Fileobj=async_buffer
+                )
+                source_file_name = async_buffer.name
+            await asyncio.to_thread(
+                convert_to_b64,
+                source=source_file_name,
+            )
+            async with aiofiles.open(file=source_file_name, mode="r") as async_buffer:
+                document["_attachment"] = (await async_buffer.read()).strip()
+            await remove(source_file_name)  # pyright: ignore
```

Comment:
this needs to go to a finally section, so you delete the file if present, whatever was the error